lighthouse-core/config/budget.js

```diff
@@ -96,6 +98,97 @@ class Budget {
     };
   }
 
+  /**
+   * Validates that path is properly formed. Verifies the quantity and location
+   * of the two robot.txt regex characters: $, *
+   * @param {unknown} val
+   * @return {string}
+   */
+  static validatePath(val) {
+    /*
+     * If no path is specified the budget is assumed to apply to all pages. This makes
+     * budget.json backwards compatible with earlier API versions that did not include path.
+     */
+
+    if ((val === undefined) || (val === null)) {
+      return '/';
+    }
+    const path = /** @type {string} */ (val);
+    const hasLeadingSlash = path.startsWith('/');
+    const validWildcardQuantity = ((path.match(/\*/g) || []).length <= 1);
+    const validDollarSignQuantity = ((path.match(/\$/g) || []).length <= 1);
+    const validDollarSignPlacement = !path.includes('$') || path.endsWith('$');
```

Comment:
~thoughts on just checking this with the single regex? i.e. `path.match(/^[^$]+\$?$/)`~

I say this, but then I look at this regex and say "ew, gross", so maybe not 😆 it's really just the `!.includes` bit that made me suggest this